src/settings.rs

```diff
@@ -105,9 +109,47 @@ impl kubewarden::settings::Validatable for Settings {
             return Err("Signatures must not be empty".to_string());
         }
 
-        // TODO: when a certificate is being used, ensure the certificate
-        // can be trusted. This requires a new waPC function being written
-
+        // when a certificate is being used, ensure it can be trusted
+        for signature in self.signatures.iter() {
+            if let Signature::Certificate(s) = signature {
+                // build sdk structs:
+                let cert = SDKCert {
+                    encoding: CertificateEncoding::Pem,
+                    data: s.certificate.clone().into_bytes(),
+                };
+                let cert_chain_opt: Option<Vec<SDKCert>> = match &s.certificate_chain {
+                    Some(chain_vec) => {
+                        // build vec of sdk certs:
+                        let mut chain_sdk: Vec<SDKCert> = vec![];
+                        chain_vec.iter().for_each(|d| {
+                            chain_sdk.push(SDKCert {
+                                encoding: CertificateEncoding::Pem,
+                                data: d.clone().into_bytes(),
+                            });
+                        });
+                        Some(chain_sdk)
+                    }
+                    None => None,
+                };
+                match verify_cert(cert, cert_chain_opt, Some(Utc::now().to_rfc3339())) {
+                    Ok(verified) => {
+                        if !verified {
+                            return Err(format!(
```

Comment:
This will never happen tbf. The `verify_cert()` function either returns `Ok<true>` or `Err(e)`.

Reply:
This is my fault, I didn't realize that while reviewing the change to the SDK. I would prefer to have the SDK do something like `Ok<()>`.

Reply:
My fault too, didn't realize when writing.

Amended all the stack, so:
* the Response returns:
  ```terraform
  trusted: bool,
  reason: string // optional, used when trusted == false
  ```
  See https://github.com/kubewarden/docs/pull/154
* SDK's  `verify_cert()` returns `Result<SDK::BoolWithReason>`
  https://github.com/kubewarden/policy-sdk-rust/pull/72
* policy-evalutator's `verify_certificate()` returns `Result<policy-evaluator::BoolWithReason>`
  https://github.com/kubewarden/policy-evaluator/pull/214